### PR TITLE
Add path-based data handler route

### PR DESCRIPTION
**Pull request by anacrolix**
Improve data handler description in README and let GoLand clobber markdownfmt's formatting, which doesn't seem to handle multiple paragraphs in a single dot point.

Improve abstractions around torrent context, common query keys and content-disposition handling.

Fixes https://github.com/anacrolix/confluence/issues/17.

Alternative to https://github.com/anacrolix/confluence/pull/33.

### CODE REVIEW

File: confluence/middlewares.go [hunk 1]
```diff
@@ -98,6 +110,27 @@ func (me *Handler) withTorrentContext(h func(w http.ResponseWriter, r *request))
 	})
 }
 
+func (me *Handler) withTorrentContextFromPath(h func(http.ResponseWriter, *request)) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		me.withTorrentContext(h, func() (ih metainfo.Hash, err error, afterAdd func(t *torrent.Torrent)) {
+			p := r.URL.Path
+			start := 1 // path should always start with /
```
Comment (ftk) [inline confluence/middlewares.go:117]: It seems like it needs to check if the path starts with "/infohash/"...

Comment (anacrolix) [reply confluence/middlewares.go:117]: Ah yes I forgot that bit.

Comment (anacrolix) [reply confluence/middlewares.go:117]: Take another look.

**Comment by ftk** [comment]
Been using it for a few days, works fine

**Pull request merged**